aws/rust-runtime/aws-runtime/src/auth/sigv4.rs

```diff
@@ -146,76 +145,47 @@ impl SigV4Signer {
 
     /// Signs the given `request`.
     ///
-    /// This is a helper used by [`Sign::sign_http_request`] and will be useful if calling code
-    /// needs to pass a configured `settings`.
+    /// This is a helper used by `(SigV4Signer as Sign).sign_http_request(...)` and will be useful
+    /// if calling code needs to pass a configured [`SigningSettings`].
     ///
-    /// TODO(S3Express): Make this method more user friendly, possibly returning a builder
-    ///  instead of taking these input parameters. The builder will have a `sign` method that
-    ///  does what this method body currently does.
+    /// If you already have an [`HttpRequest`] and a [`SigningParams`] at hand, consider using
+    /// [`SigningApplication`] directly.
     pub fn sign_http_request(
```

Comment:
We can't change the arguments on this function since it's part of the public API.

Reply:
Ah, this is a new function only existing in my feature branch. The semver failure in CI is both right and wrong; it's right because the arguments changed in this PR but wrong because it's not a semver failure w.r.t the `main` branch.

Reply:
Oh, right. I forgot.